storage/src/persistent/mod.rs

```diff
@@ -153,24 +156,32 @@ impl PersistentStorage {
         clog: Arc<CommitLogs>,
         merkle_backend: KeyValueStoreBackend,
     ) -> Self {
-        let merkle = match merkle_backend {
-            KeyValueStoreBackend::RocksDB => MerkleStorage::new(Box::new(RocksDBBackend::new(
-                db_context.clone(),
-                MerkleStorage::name(),
-            ))),
-            KeyValueStoreBackend::InMem => MerkleStorage::new(Box::new(InMemoryBackend::new())),
-            KeyValueStoreBackend::Sled { path } => {
-                let sled = sled::Config::new().path(path).open().unwrap();
-                MerkleStorage::new(Box::new(SledBackend::new(sled.deref().clone())))
-            }
-            KeyValueStoreBackend::BTreeMap => MerkleStorage::new(Box::new(BTreeMapBackend::new())),
-        };
+        let merkle =
+            match merkle_backend {
+                KeyValueStoreBackend::RocksDB => {
+                    MerkleStorage::new(Box::new(RocksDBBackend::new(db_context.clone())))
+                }
+                KeyValueStoreBackend::InMem => MerkleStorage::new(Box::new(InMemoryBackend::new())),
+                KeyValueStoreBackend::Sled { path } => {
+                    let sled = sled::Config::new().path(path).open().unwrap();
```

Comment:
Is there anything that can be done about this `unwrap` here?

Reply:
@mat13mn this one was marked as resolved but I don't see any change. Since the return type of this function is `Result<...>`, I think this can be mapped to an error.

Reply:
ouch - i changed the return type but forgot to remove unwrap ...